contracts/utils/convex/ConvexYieldWrapper.sol

```diff
@@ -61,6 +60,20 @@ contract ConvexStakingWrapperYield is ConvexStakingWrapper {
         emit VaultSet(account, vaultId);
     }
 
+    /// @notice Remove a vault from the user's vault list
+    /// @param vaultId The vaulId being added
+    /// @param account The user from whom the vault needs to be removed
+    function removeVault(bytes12 vaultId, address account) public {
```

Comment:
To make this permissionless you should pull the `owner` of the vault. If `account != owner` then you remove the vault from `vaults`.

The logic is this:
If a user owns a vault, anyone can register that vault for that owner in the wrapper.
If a user doesn't own a vault, but it is registered for him in the wrapper, anyone can remove it.

Removing vaults is only needed to save a bit of gas on `_getDepositedBalance`, so it doesn't really matter if users sometimes have vaults that don't belong to them. They will be ignored.